examples/performance/main.rs

```diff
@@ -89,100 +91,143 @@ trait Renderer: Drop {
     fn window(&mut self) -> &glutin::Window;
 }
 
-
-
 struct GFX {
     dimension: i16,
-    window: glutin::Window,
-    device:gfx_device_gl::Device,
-    encoder: gfx::Encoder<R,CB>,
+    window: gfx_window_glutin::Window,
+    surface: gfx_window_glutin::Surface,
+    swap_chain: gfx_window_glutin::SwapChain,
+    queue: gfx::queue::GraphicsQueue<B>,
+    pool: gfx::GraphicsCommandPool<B>,
+    frame_semaphore: gfx::handle::Semaphore<R>,
+    draw_semaphore: gfx::handle::Semaphore<R>,
+    views: Vec<gfx::handle::RenderTargetView<R, ColorFormat>>,
     data: pipe::Data<R>,
     pso: gfx::PipelineState<R, pipe::Meta>,
     slice: gfx::Slice<R>,
 }
 
-struct GL {
-    dimension: i16,
-    window: glutin::Window,
-    gl:Gl,
-    trans_uniform:GLint,
-    vs:GLuint,
-    fs:GLuint,
-    program:GLuint,
-    vbo:GLuint,
-    vao:GLuint,
-}
-
-
 impl GFX {
     fn new(builder: glutin::WindowBuilder, events_loop: &glutin::EventsLoop, dimension: i16) -> Self {
         use gfx::traits::FactoryExt;
 
-        let (window, device, mut factory, main_color, _) =
-            gfx_window_glutin::init::<ColorFormat, DepthStencil>(builder, events_loop);
-        let encoder: gfx::Encoder<_,_> = factory.create_command_buffer().into();
+        // Create window
+        let events_loop = glutin::EventsLoop::new();
+        let builder = glutin::WindowBuilder::new()
+            .with_title("Triangle example".to_string())
+            .with_dimensions(1024, 768)
+            .with_vsync();
+        let win = gfx_window_glutin::build(builder, &events_loop, ColorFormat::get_format(), DepthStencil::get_format());
+        let mut window = gfx_window_glutin::Window::new(win);
+        // Acquire surface and adapters
+        let (mut surface, adapters) = window.get_surface_and_adapters();
+        let queue_descs = adapters[0].get_queue_families().iter()
+                                    .filter(|family| surface.supports_queue(&family) )
+                                    .map(|family| { (family, 1) })
+                                    .collect::<Vec<_>>();
+
+        // Open device (factory and queues)
+        let gfx::Device { mut factory, mut general_queues, mut graphics_queues, .. } = adapters[0].open(&queue_descs);
+        let mut queue = if let Some(queue) = general_queues.pop() {
```

Comment:
what I was thinking is having a wrapper enum (that is either general or graphics queue) that is dereffable to the graphics one. In this case, that giant `if/else` section could be omitted